rct229/rules/section6/section6rule8.py

```diff
@@ -0,0 +1,128 @@
+            list_path="ruleset_model_instances[0]",
+            required_fields={"$": ["calendar"], "calendar": ["is_leap_year"]},
+        )
+
+    def create_data(self, context, data=None):
+        rmd_b = context.baseline
+        return {"is_leap_year": rmd_b["calendar"]["is_leap_year"]}
+
+    class RuleSetModelInstanceRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section6Rule8.RuleSetModelInstanceRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, True),
+                each_rule=Section6Rule8.RuleSetModelInstanceRule.BuildingRule(),
+                index_rmr="proposed",
+                list_path="buildings[*]",
+                required_fields={"$": ["schedules"]},
+            )
+
+        def create_data(self, context, data=None):
+            rmd_b = context.baseline
+            rmd_p = context.proposed
+            return {
+                **data,
+                "schedules_b": rmd_b["schedules"],
+                "schedules_p": rmd_p["schedules"],
+            }
+
+        class BuildingRule(RuleDefinitionListIndexedBase):
+            def __init__(self):
+                super(Section6Rule8.RuleSetModelInstanceRule.BuildingRule, self).__init__(
+                    rmrs_used=UserBaselineProposedVals(False, True, True),
+                    each_rule=Section6Rule8.RuleSetModelInstanceRule.BuildingRule.ZoneRule(),
+                    index_rmr="proposed",
+                    list_path="$..zones[*]",
+                )
+
+            def create_data(self, context, data=None):
+                building_p = context.proposed
+                schedules_p = data["schedules_p"]
+                return {
+                    **data,
+                    "building_open_schedule_p":  getattr_(
+                        find_one_with_field_value("$", "id", building_p["building_open_schedule"], schedules_p),
+                        "schedule", "hourly_values")
+                }
+
+            class ZoneRule(RuleDefinitionListIndexedBase):
+                def __init__(self):
+                    super(Section6Rule8.RuleSetModelInstanceRule.BuildingRule.ZoneRule, self).__init__(
+                        rmrs_used=UserBaselineProposedVals(False, True, True),
+                        each_rule=Section6Rule8.RuleSetModelInstanceRule.BuildingRule.ZoneRule.SpaceRule(),
+                        index_rmr="proposed",
+                        list_path="spaces[*]",
+                        required_fields={
+                            "$": ["volume"]
+                        }
+                    )
+
+                def create_data(self, context, data=None):
+                    zone_p = context.proposed
+                    return {
+                        **data,
+                        "avg_space_height": zone_p.get("volume", 0.0) / pint_sum(find_all("spaces[*].floor_area", zone_p), ZERO.AREA)
```

Comment:
It turns out that we don't need `pint_sum`.  `sum` works for pint Quantities directly.